src/persist/src/indexed/columnar.rs

```diff
@@ -438,4 +564,31 @@ mod tests {
             .collect();
         assert_eq!(reads, updates);
     }
+
+    #[test]
+    fn vec_builder() {
+        fn testcase(max_len: usize, key_size: usize, num_records: usize) {
+            let key = "x".repeat(key_size);
+            let expected = (0..num_records)
+                .map(|x| ((key.as_bytes(), "".as_bytes()), u64::cast_from(x), 1))
+                .collect::<Vec<_>>();
+            let mut builder = ColumnarRecordsVecBuilder::new_with_len(max_len);
+            // Call reserve once at the beginning to match the usage in the
+            // FromIterator impls.
+            builder.reserve(num_records, key.len(), 0);
+            for (idx, x) in expected.iter().enumerate() {
+                builder.push(*x);
+                assert_eq!(builder.len(), idx + 1);
+            }
+            let columnar = builder.finish();
+            let actual = columnar.iter().flat_map(|x| x.iter()).collect::<Vec<_>>();
+            assert_eq!(actual, expected);
+        }
+
+        testcase(KEY_VAL_DATA_MAX_LEN, 0, 0);
+        testcase(KEY_VAL_DATA_MAX_LEN, 10, 10);
+        testcase(10, 10, 1);
```

Comment:
nit: I know for sure that this is mildly annoying so feel free to ignore but i feel like its worth it to have a testcase that validates that `testcase(10, 11, 1)` panics

Reply:
nah i think panic testing is great. however, this would require either a 2GB alloc in a unit test or plumbing the `limit` thing down to ColumnarRecordBuilder, both of which I'd rather not do. I'll do the latter if you insist, though

Reply:
nah thats fine